bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/handler/DoorbellHandler.java

```diff
@@ -458,56 +448,63 @@ private void updateMontage(String event, String channelId) {
+                DoorbirdImage historyImage = CHANNEL_DOORBELL_IMAGE_MONTAGE.equals(channelId)
+                        ? api.downloadDoorbellHistoryImage(String.valueOf(imageNumber))
+                        : api.downloadMotionHistoryImage(String.valueOf(imageNumber));
+                if (historyImage != null) {
+                    RawType image = historyImage.getImage();
+                    if (image != null) {
+                        try {
+                            BufferedImage i = ImageIO.read(new ByteArrayInputStream(image.getBytes()));
+                            images.add(i);
+                        } catch (IOException e) {
+                            logger.debug("IOException creating BufferedImage from downloaded image: {}",
+                                    e.getMessage());
+                        }
                     }
                 }
             }
-        }
-        if (images.size() < numberOfImages) {
-            logger.debug("Some images could not be downloaded: wanted={}, actual={}", numberOfImages, images.size());
+            if (images.size() < numberOfImages) {
+                logger.debug("Some images could not be downloaded: wanted={}, actual={}", numberOfImages,
+                        images.size());
+            }
         }
         return images;
     }
 
     // Assemble the array of images into a single scaled image
     private @Nullable State createMontage(ArrayList<BufferedImage> images) {
-        int height = (int) (images.get(0).getHeight() * (config.montageScaleFactor / 100.0));
-        int width = (int) (images.get(0).getWidth() * (config.montageScaleFactor / 100.0));
-        int widthTotal = width * images.size();
-        logger.debug("Dimensions of final montage image: w={}, h={}", widthTotal, height);
-
-        // Create concatenated image
-        int currentWidth = 0;
-        BufferedImage concatImage = new BufferedImage(widthTotal, height, BufferedImage.TYPE_INT_RGB);
-        Graphics2D g2d = concatImage.createGraphics();
-        logger.debug("Concatenating images array into single image");
-        for (int j = 0; j < images.size(); j++) {
-            g2d.drawImage(images.get(j), currentWidth, 0, width, height, null);
-            currentWidth += width;
-        }
-        g2d.dispose();
-
-        // Convert image to a state
-        logger.debug("Rendering image to byte array and converting to RawType state");
-        byte[] imageBytes = convertImageToByteArray(concatImage);
         State state = null;
-        if (imageBytes != null) {
-            state = new RawType(imageBytes, "image/png");
+        Integer montageScaleFactor = config.montageScaleFactor;
+        if (montageScaleFactor != null) {
+            int height = (int) (images.get(0).getHeight() * (montageScaleFactor / 100.0));
+            int width = (int) (images.get(0).getWidth() * (montageScaleFactor / 100.0));
+            int widthTotal = width * images.size();
```

Comment:
How do you know that all the images will have an identical size?

Reply:
I made this assumption because the Doorbird images are a fixed resolution that cannot be changed by the user. At a minimum, I probably should address this in a comment.